configuration-baselines/.htaccess-joomla

```diff
@@ -0,0 +1,192 @@
+##
+# @copyright  Copyright (C) 2005 - 2020 Open Source Matters. All rights reserved.
+# @license    GNU General Public License version 2 or later; see LICENSE.txt
+##
+
+###########################################
+# ========= No directory listings =========
+IndexIgnore *
+Options +FollowSymLinks
+Options -Indexes
+###########################################
+
+###########################################
+# ======= Enable the Rewrite Engine =======
+RewriteEngine On
+###########################################
+
+###########################################
+# ==== Block out some common exploits. ====
+RewriteCond %{QUERY_STRING} base64_encode[^(]*\([^)]*\) [OR]
+RewriteCond %{QUERY_STRING} (<|%3C)([^s]*s)+cript.*(>|%3E) [NC,OR]
+RewriteCond %{QUERY_STRING} GLOBALS(=|\[|\%[0-9A-Z]{0,2}) [OR]
+RewriteCond %{QUERY_STRING} _REQUEST(=|\[|\%[0-9A-Z]{0,2})
+RewriteRule .* index.php [F]
+###########################################
+
+###########################################
+# ======= Joomla! core SEF Section. =======
+RewriteRule .* - [E=HTTP_AUTHORIZATION:%{HTTP:Authorization}]
+RewriteCond %{REQUEST_URI} !^/index\.php
+RewriteCond %{REQUEST_FILENAME} !-f
+RewriteCond %{REQUEST_FILENAME} !-d
+RewriteRule .* index.php [L]
+###########################################
+
+###########################################
+# === File execution order ===
+DirectoryIndex index.php index.html
+###########################################
+
+###########################################
+# ========== Add redirects here ===========
+
+###########################################
+
+###########################################
+# === Optimal default expiration time ===
+<IfModule mod_expires.c>
```

Comment:
The `mod_expires`, `mod_deflate`, and `mod_gzip` stuff can be in the general config.  These sections are dealing with defining "optimal" (opinionated) cache timeframes for different asset types and optimizing resource delivery.

The stuff about not caching `administrator/index.php` or rewrite rules for the CMS filesystem structure or query strings for the CMS' non-SEF configuration is what rightfully belongs in this file.